File: software/examples/java/ExampleLoadImage.java
import java.awt.image.BufferedImage;
import javax.imageio.ImageIO;
import java.io.File;
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletEPaper296x128;

public class ExampleLoadImage {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XZY"; // Change XYZ to the UID of your E-Paper 296x128 Bricklet
	private static final short WIDTH = 296;
	private static final short HEIGHT = 128;

	static boolean[] boolArrayFromImage(BufferedImage image, int color) throws Exception {
		boolean[] pixels = new boolean[HEIGHT * WIDTH];
		short h, w;

		for (h = 0; h < HEIGHT; h++) {
			for (w = 0; w < WIDTH; w++) {
				pixels[h*WIDTH + w] = ((image.getRGB(w, h) & 0x00FFFFFF) == color);
			}
		}

		return pixels;
	}

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletEPaper296x128 epaper = new BrickletEPaper296x128(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Download example image here:
		// https://raw.githubusercontent.com/Tinkerforge/e-paper-296x128-bricklet/master/software/examples/tf_red.png
		BufferedImage image = ImageIO.read(new File("tf_red.png"));

		// Get black/white pixels from image and write them to the Bricklet buffer
		boolean[] pixelsBW = boolArrayFromImage(image, 0xFFFFFF);
		epaper.writeBlackWhite(0, 0, WIDTH-1, HEIGHT-1, pixelsBW);

		// Get red pixels from image and write them to the Bricklet buffer
		boolean[] pixelsRed = boolArrayFromImage(image, 0xFF0000);
		epaper.writeColor(0, 0, WIDTH-1, HEIGHT-1, pixelsRed);

		// Draw buffered values to the display
		epaper.draw();
	}
}

File: software/examples/php/ExampleLoadImage.php
<?php

if (!extension_loaded('gd')) {
    echo "Required gd extension is not available\n";
    exit;
}

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletEPaper296x128.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletEPaper296x128;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your E-Paper 296x128 Bricklet
const WIDTH = 296; // Columns
const HEIGHT = 128; // Rows

// Convert GD Image to matching color bool array
function boolArrayFromImage($image, $r, $g, $b)
{
    $pixels = array();

    // Convert image pixels into 8bit pages
    for ($row = 0; $row < HEIGHT; $row++)
    {
        for ($column = 0; $column < WIDTH; $column++)
        {
            $index = imagecolorat($image, $column, $row);
			$color = imagecolorsforindex($image, $index);
	        $pixels[$row*WIDTH + $column] = (($color['red'] == $r) && ($color['green'] == $g) && ($color['blue'] == $b));
        }
    }
    
	return $pixels;
}

$ipcon = new IPConnection(); // Create IP connection
$epaper = new BrickletEPaper296x128(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Download example image here:
// https://raw.githubusercontent.com/Tinkerforge/e-paper-296x128-bricklet/master/software/examples/tf_red.png
$image = imagecreatefrompng('tf_red.png');

// Get black/white pixels from image and write them to the Bricklet buffer
$pixelsBW = boolArrayFromImage($image, 0xFF, 0xFF, 0xFF);
$epaper->writeBlackWhite(0, 0, WIDTH-1, HEIGHT-1, $pixelsBW);

// Get red pixels from image and write them to the Bricklet buffer
$pixelsRed = boolArrayFromImage($image, 0xFF, 0, 0);
$epaper->writeColor(0, 0, WIDTH-1, HEIGHT-1, $pixelsRed);

// Draw buffered values to the display
$epaper->draw();

echo "Press ctrl+c to exit\n";

?>
